plugins/inputs/fluentd/fluentd.go

```diff
@@ -0,0 +1,195 @@
+package fluentd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	measurement  = "fluentd"
+	description  = "Read metrics exposed by fluentd in_monitor plugin"
+	sampleConfig = `
+	## This plugin only reads information exposed by fluentd using /api/plugins.json.
+	## Tested using 'fluentd' version '0.14.9'
+	##
+	## Endpoint: 
+	## - only one URI is allowed
+	## - https is not supported
+	endpoint = "http://localhost:24220/api/plugins.json"
+	
+	## Define which plugins have to be excluded (based on "type" field - e.g. monitor_agent)
+	exclude = [
+		"monitor_agent",
+		"dummy",
+	]
+`
+)
+
+// Fluentd - plugin main structure
+type Fluentd struct {
+	Endpoint string
+	Exclude  []string
+	client   *http.Client
+}
+
+type pluginData struct {
+	PluginID              string  `json:"plugin_id"`
+	PluginType            string  `json:"type"`
+	PluginCategory        string  `json:"plugin_category"`
+	RetryCount            float64 `json:"retry_count"`
+	BufferQueueLength     float64 `json:"buffer_queue_length"`
+	BufferTotalQueuedSize float64 `json:"buffer_total_queued_size"`
+}
+
+// parse JSON from fluentd Endpoint
+// Parameters:
+// 		data: unprocessed json recivied from endpoint
+//
+// Returns:
+//		pluginData:		slice that contains parsed plugins
+//		error:			error that may have occurred
+func parse(data []byte) ([]pluginData, error) {
+	var (
+		pdPoint      pluginData
+		pdPointArray []pluginData
+		parsed       map[string]interface{}
```

Comment:
In this case it may be better to use pointers in the struct and Unmarshal directly into it.  If the field is not in the JSON it will be nil in the struct.